firewall/firewall_test.go

```diff
@@ -434,11 +435,135 @@ func testBlacklistByRules(t *testing.T, firewall * Firewall, blacklistHandler *
 	for _, query := range testQueries {
 		err := firewall.HandleQuery(query)
 		if err != nil {
-			if err != handlers.ErrForbiddenSqlStructure {
+			if err != handlers.ErrForbiddenSqlStructureBlacklist {
+				t.Fatal(err)
+			}
+		} else {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestConfigurationProvider(t *testing.T) {
+
+	var DEFAULT_CONFIG_PATH = utils.GetConfigPathByName("acra_firewall.example")
+
+	filePath, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	configuration, err := ioutil.ReadFile(filepath.Join(filePath, "../", DEFAULT_CONFIG_PATH))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	firewall := &Firewall{}
+
+	err = firewall.SetFirewallConfiguration(configuration)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testQueries := []string {
+		"INSERT INTO SalesStaff1 VALUES (1, 'Stephen', 'Jiang');",
+		"SELECT AVG(Price) FROM Products;",
+	}
+
+	//firewall should block those queries (blacklist works)
+	for _, queryToBlock := range testQueries{
+		err = firewall.HandleQuery(queryToBlock)
+		if err != nil {
+			if err != handlers.ErrQueryInBlacklist{
+				t.Fatal(err)
+			}
+		} else {
+			t.Fatal(err)
+		}
+	}
+
+	testQueries = []string {
+		"INSERT INTO EMPLOYEE_TBL VALUES (1, 'Stephen', 'Jiang');",
+		"SELECT AVG(Price) FROM Customers;",
+	}
+
+	//firewall should block those tables (blacklist works)
+	for _, queryToBlock := range testQueries{
+		err = firewall.HandleQuery(queryToBlock)
+		if err != nil {
+			if err != handlers.ErrAccessToForbiddenTableBlacklist {
+				t.Fatal(err)
+			}
+		} else {
+			t.Fatal(err)
+		}
+	}
+
+	testQueries = []string {
+		"SELECT EMP_ID, LAST_NAME FROM EMPLOYEE WHERE CITY = 'Seattle' ORDER BY EMP_ID;",
+		"SELECT EMP_ID, LAST_NAME FROM EMPLOYEE AS EMPL WHERE CITY = 'Seattle' ORDER BY EMP_ID;",
+	}
+
+	//firewall should block those structures (blacklist works)
+	for _, queryToBlock := range testQueries{
+		err = firewall.HandleQuery(queryToBlock)
+		if err != nil {
+			if err != handlers.ErrForbiddenSqlStructureBlacklist {
+				t.Fatal(err)
+			}
+		} else {
+			t.Fatal(err)
+		}
+	}
+
+	testQueries = []string {
+		"SELECT EMP_ID, LAST_NAME FROM PRODUCTS WHERE CITY='INDIANAPOLIS' ORDER BY EMP_ID;",
+		"SELECT EMP_ID, LAST_NAME FROM PRODUCTS WHERE CITY='INDIANAPOLIS' ORDER BY EMP_ID asc;",
+	}
+
+	//firewall should block those tables (whitelist works)
+	for _, queryToBlock := range testQueries{
+		err = firewall.HandleQuery(queryToBlock)
+		if err != nil {
+			if err != handlers.ErrAccessToForbiddenTableWhitelist {
 				t.Fatal(err)
 			}
 		} else {
 			t.Fatal(err)
 		}
 	}
+
+
+	testSyntax(t)
 }
+func testSyntax(t *testing.T) {
```

Comment:
better to rename `TestLoadConfiguration` because when I read first, I thought that we test here correct parsing of yaml format. But we don't need it, because it tested by yaml parser itself

Reply:
No. yaml parser can't check those error inputs : "SLECT AVG(Price) FROM Products;" or "SELECT * ROM EMPLOYEE WHERE CITY='Seattle';". These syntax errors can be checked only by sqlparser